pkg/channel/producer/producer.go

```diff
@@ -86,6 +89,12 @@ func (p *Producer) ProduceKafkaMessage(ctx context.Context, channelReference eve
 	topicName := util.TopicName(channelReference)
 	logger := p.logger.With(zap.String("Topic", topicName))
 
+	// Append Time Transformer To Add "ce_time" Extension If Not Present
+	transformers = append(transformers, transformer.AddExtension("time", time.Now()))
```

Comment:
You should not mutate the event for any reason (except the knativehistory header), look here: https://github.com/knative/eventing/blob/master/docs/spec/channel.md#output The imc channel used to add the `time` attribute, but it was removed.